.changeset/ten-hats-kick.md

```diff
@@ -0,0 +1,10 @@
+---
+'@backstage/plugin-badges-backend': patch
+'@backstage/plugin-catalog-backend': patch
+'@backstage/plugin-jenkins-backend': patch
+'@backstage/plugin-search-backend': patch
+'@backstage/plugin-techdocs-backend': patch
+'@backstage/plugin-todo-backend': patch
+---
+
+Make use of the new `IdentityApi` interface to retrieve the user token.
```

Comment:
I think at this point it's worthwhile making separate changesets for these with a little diff too, with instructions for how to add the identity to each one in `packages/backend`. That way we can get some movement toward people actually making use of the new class out there

Reply:
As I start to integrate it with the example app, I see some errors. Looks as though backend to backend comms is failing..

The DefaultIdentityClient, does not appear to accept backend to backend tokens. I think I need a lead on this one..

I am going to see how I can get the DefaultIdentityClient to accept backend tokens, but I'd love thoughts on this issue.

```
2022-09-02T07:24:33.816Z backstage info ::ffff:127.0.0.1 - - [02/Sep/2022:07:24:33 +0000] "GET /api/catalog/entities?offset=0&limit=500 HTTP/1.1" 403 727 "-" "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)" type=incomingRequest
2022-09-02T07:24:33.821Z search error Collating documents for software-catalog failed: ResponseError: Request failed with 403 NotAllowedError type=plugin
2022-09-02T07:24:33.823Z search error Request failed with 403 NotAllowedError type=taskManager task=search_index_software_catalog name=ResponseError response=[object Response] body=[object Object] cause=NotAllowedError: "alg" (Algorithm) Header Parameter not allowed stack=ResponseError: Request failed with 403 NotAllowedError
    at Function.fromResponse (webpack-internal:///../errors/src/errors/ResponseError.ts:42:16)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async CatalogClient.requestRequired (webpack-internal:///../catalog-client/src/CatalogClient.ts:248:19)
    at async CatalogClient.getEntities (webpack-internal:///../catalog-client/src/CatalogClient.ts:86:26)
    at async DefaultCatalogCollatorFactory.execute (webpack-internal:///../../plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts:52:31)
    at async next (node:internal/streams/from:85:11)

```

Reply:
I suspect there might be a race condition in the ServerTokenManager, where you have a 2 getToken calls and an authenticate call generated quickly before any signing keys or during the creation of a signing key.

Reply:
done